client/packages/common/src/utils/regex/RegexUtils.tsx

```diff
@@ -44,6 +45,20 @@ export const RegexUtils = {
       RegexUtils.includes(substring, String(object[key]))
     );
   },
-  escapeChars: (regexString?: string) =>
-    regexString?.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'),
+  escapeChars: (regexString: string) =>
+    regexString.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'),
+
+  // Takes a string formatted in template-literal style (i.e. using ${ }) and
+  // replaces contents of the matching parameters with values from provided
+  // data object.
+  // Eg. "My name is ${user.name}" with object {user: {name: "Frodo"}}
+  // returns "My name is Frodo"
+  stringSubstitution: (
+    parameterisedString: string,
+    data: { [key: string]: any },
+    fallback = 'Not found'
+  ) =>
+    parameterisedString.replace(/\${(.*?)}/gm, (_: string, match: string) =>
+      extractProperty(data, match, fallback)
+    ),
```

Comment:
A little copy-paste from Conforma ;)

Reply:
Nice!

and.. oh, right. the name `stringSubstitution` referenced earlier did not make me expect this kind of functionality. Something like 'formatTemplateString' or 'templateSubstitution' 🤷 dunno